tensorpipe/common/cuda_loop.cc

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include <tensorpipe/common/cuda_loop.h>
+
+#include <tensorpipe/common/cuda.h>
+#include <tensorpipe/common/system.h>
+
+namespace tensorpipe {
+
+CudaLoop::CudaLoop() {
+  std::unique_lock<std::mutex> lock(mutex_);
+  thread_ = std::thread([this, lock{std::move(lock)}]() mutable {
+    setThreadName("TP_CUDA_callback_loop");
+    for (;;) {
+      if (operations_.empty()) {
+        if (joined_ && pendingOperations_ == 0) {
+          break;
+        } else {
+          cv_.wait(lock);
```

Comment:
Nit: some linters suggest to always pass a predicate to `wait()` so that the stdlib will save you from spurious waits or from going to sleep when you could be doing something. In this case the condition is actually somewhat complicated, so perhaps doing it that way would be less readable. You decide.